scripts/test/ReductionWrapperTest.py

```diff
@@ -60,116 +61,98 @@ def reduce(self, input_file = None, output_directory = None):
-
+        red = ReductionWrapper('MAR')
 
-        self.assertRaises(NotImplementedError,red.def_main_properties)
-        self.assertRaises(NotImplementedError,red.def_advanced_properties)
+        self.assertRaises(NotImplementedError, red.def_main_properties)
+        self.assertRaises(NotImplementedError, red.def_advanced_properties)
         self.assertTrue('reduce' in dir(red))
 
     def test_export_advanced_values(self):
         red = mr.ReduceMARI()
 
-        main_prop=red.def_main_properties()
-        adv_prop=red.def_advanced_properties()
+        main_prop = red.def_main_properties()
+        adv_prop = red.def_advanced_properties()
 
-
-        # see what have changed and what have changed as advanced properties. 
+        # see what have changed and what have changed as advanced properties.
         all_changed_prop = red.reducer.prop_man.getChangedProperties()
 
-        self.assertEqual(set(list(main_prop.keys())+list(adv_prop.keys())),all_changed_prop)
-
-        test_dir = config['defaultsave.directory']
-        file = os.path.join(test_dir,'reduce_vars.py')
-        #clear up previous rubbish may be present from other runs
-        if os.path.isfile(file):
-            os.remove(file)
-        fbase,fext = os.path.splitext(file)
-        fcomp = fbase+'.pyc'
-        if os.path.isfile(fcomp):
-            os.remove(fcomp)
-        # save web variables
-        red.save_web_variables(file)
-        self.assertTrue(os.path.isfile(file))
+        self.assertEqual(set(list(main_prop.keys()) + list(adv_prop.keys())), all_changed_prop)
 
-        # restore saved parameters.
-        sys.path.insert(0,test_dir)
+        with TemporaryDirectory() as reduce_vars_dir:
+            reduce_vars_file = os.path.join(reduce_vars_dir, 'reduce_vars.py')
+            # save web variables
+            red.save_web_variables(reduce_vars_file)
+            self.assertTrue(os.path.isfile(reduce_vars_file))
 
-        import reduce_vars as rv
+            # restore saved parameters.
+            sys.path.insert(0, reduce_vars_dir)
 
-        self.assertDictEqual(rv.standard_vars,main_prop)
-        self.assertDictEqual(rv.advanced_vars,adv_prop)
-        self.assertTrue(hasattr(rv,'variable_help'))
+            import reduce_vars as rv
 
-        imp.reload(mr)
+            self.assertDictEqual(rv.standard_vars, main_prop)
+            self.assertDictEqual(rv.advanced_vars, adv_prop)
+            self.assertTrue(hasattr(rv, 'variable_help'))
 
-        # tis will run MARI reduction, which probably not work from unit tests
-        # will move this to system tests
-        #rez = mr.main()
+            imp.reload(mr)
 
-        self.assertTrue(mr.web_var)
-        self.assertEqual(mr.web_var.standard_vars,main_prop)
-        self.assertEqual(mr.web_var.advanced_vars,adv_prop)
+            # tis will run MARI reduction, which probably not work from unit tests
+            # will move this to system tests
+            #rez = mr.main()
```

Comment:
Just a nit-pick that isn't worth holding this up with the breakage on Master, do we have an issue to move this out so it doesn't end up forgotten?